python/hsfs/engine/spark.py

```diff
@@ -869,6 +871,22 @@ def _apply_transformation_function(self, training_dataset, dataset):
         dataset = dataset.select(*sorted_feature_names)
         return dataset
 
+    def _time_series_split(self, training_dataset, dataset, split_weights):
+        feature_dataframe_list = []
+        dataset = dataset.withColumn(
+            "rank",
+            percent_rank().over(
+                window.Window.partitionBy().orderBy(
+                    training_dataset._querydto._left_feature_group.event_time
+                )
+            ),
+        )
+        for split_weight in split_weights:
+            feature_dataframe_list.append(
+                dataset.where("rank <= {:.2f}".format(split_weight)).drop("rank")
```

Comment:
rank should be in range `[accumulate_weight, accumulate_weight+ split_weight]`?

Reply:
ahh yes. good point